src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GT4Entity_ThermalBoiler.java

```diff
@@ -143,6 +143,12 @@ public boolean checkRecipe(ItemStack aStack) {
                                 this.mOutputItems = new ItemStack[] { GT_Utility.copy(tRecipe.getOutput(5)) };
                             }
                         }
+                        if (tRecipe.mFluidOutputs.length != 0) {
+                            for (FluidStack aOutput : tRecipe.mFluidOutputs) {
+                                this.mOutputFluids = new FluidStack[] {
+                                        FluidUtils.getFluidStack(aOutput, aOutput.amount) };
+                            }
```

Comment:
Now if there is a recipe with more than 1 fluid output it will only output the last one, you should initialize the array before the loop and add outputs inside it
```suggestion
                            final FluidStack[] mFluidOutputs = tRecipe.mFluidOutputs;
                            this.mOutputFluids = new FluidStack[mFluidOutputs.length];
                            for (int i = 0, mFluidOutputsLength = mFluidOutputs.length; i < mFluidOutputsLength; i++) {
                                this.mOutputFluids[i] = mFluidOutputs[i].copy();
                            }
```

Reply:
I see what you mean now. I was trying to find examples in the rest of the mod but couldn't.